### DISCUSSION TITLE
Getting a python stage object using the Autodesk plugin

### DISCUSSION THREAD

**Discussion by pmolodo** [Autodesk/maya-usd#1264] [Q&A]
So, if you have an autodesk proxy shape, and you've built with ufe support, I know you can do:

```python
import mayaUsd.ufe
stage = mayaUsd.ufe.getStage('|dummy_ufe_root' + proxyShapeFullPath)
```

However, what if you just have a `pxrUsdStageNode`, not a proxy shape? And/or if you've compiled without ufe support?

**Comment by kxl-adsk** [comment]
We have a GetPrim query exposed to python that allows you to access the stage. It will work when you compile without UFE support, but it requires a DG node to inherit from UsdMayaUsdPrimProvider:
https://github.com/Autodesk/maya-usd/blob/91994c05e287ca22d522556eee3d82eb4c93ecc4/lib/mayaUsd/utils/query.cpp#L37

Stage node does not inherit from it, so you won't be able to use this query, unfortunately.

I'm curious about your workflow with stage nodes - can you share some more details about it?
